src/domain/disjointDomain.ml

```diff
@@ -182,16 +182,26 @@ module type MayEqualSetDomain =
 sig
   include SetDomain.S
   val may_be_equal: elt -> elt -> bool
+  val amenable_to_meet: elt -> elt -> bool
```

Comment:
I'm wondering if the two predicates could be combined into one more meaningful thing. Because I guess now there's supposed to be some kind of relationship between them, given that we only check them in certain order.

Would this be the correct description?
```ocaml
val same_sublattice: elt -> elt -> bool option
(** @return [Some true] if definitely in same sublattice, [Some false] if definitely not, [None] if unknown. *)
```

Reply:
I'm skeptical here: `may_be_equal` is a strange semantic property for which there is no real definition in terms of lattices.

So I think using lattice terminology here will be only confusing.  For example what is difference between `Some false` and `None` in your proposed definition above:
- Even if they are definitely not from the same sublattice, they may still be equal or not
- Same if I can't determine whether they are in the same sublattice.

So it once again boils down to the two properties above.

Reply:
~~~OCaml
  type eqinfo = Maybe | No | Meet
  val may_be_equal: elt -> elt -> eqinfo
~~~

One could do something like this but I don't think that is particularly nice.

Reply:
> I'm skeptical here: `may_be_equal` is a strange semantic property for which there is no real definition in terms of lattices.

I agree and that's why I'm wondering if there is a less ad-hoc way to view these things. `same_sublattice` might not be the right thing, but possibly something else.

There are now two predicates who have 4 possible combined results for a pair of arguments, but only three cases are distinguished where the predicates are used.
Suppose some addresses `x` and `y` had `may_be_equal x y = false` before. Then neither was added to the result.
Now if `amenable_to_meet x y = true`, then `meet x y` is added to the result, which makes the meet of the entire sets larger (i.e. less precise) than before.

This makes me wonder about the order of these checks. If the previous implementation was sound, then checking `may_be_equal` first (to add nothing) would still be sound and cannot become less precise due to `amenable_to_meet`, which could then be checked second to determine whether `meet x y` is added or both `x` and `y`. This would be more precise than the previous implementation by construction.

Or if the case `may_be_equal x y = false` and `amenable_to_meet x y = true` should be impossible, then why? That would indicate there being some implicit connection between the two predicates (a semantic and a lattice-theoretic) after all.